### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by caitlinrhd**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review this Pull Request and comment as if you were reviewing an intern's code.

### CODE REVIEW

File: phone_actions.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(username, password)
+    @actions_url = "https://#{username}:#{password}@api.<made_up>.com/actions/"
+  end
+
+  def call!(number, message)
+    JSON.parse(RestClient.post("@actions_url/#{text}", {message: message, number: number}))
+  end
+
+  def text!(number, message)
+    JSON.parse(RestClient.post("@actions_url/#{call}", {message: message, number: number}))
```
Comment (johnzw89) [inline phone_actions.rb:12]: See comment above

File: phone_number_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
+
+RSpec.describe PhoneNumber do
+  let(:phone_number) { PhoneNumber.new("5558675309") }
```
Comment (johnzw89) [inline phone_number_spec.rb:4]: This a great base case to test for. We should also add additional test cases. Here are some examples:
- Phone number is an int
- phone number string is already formatted "(123) 123 - 1234"
- phone number includes country code such as "+11231231234"
- phone number is not a valid phone number e.g "1234"
- what else can you think of?

File: phone_number_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    it "is the number as an int" do
+      new_number = PhoneNumber.new("555-867-5309")
+      expect(new_number.to_i).to eq(5558675309)
+    end
+  end
+
+  describe "#text" do
+    it "calls PhoneActions#text!" do
+      expect_any_instance_of(PhoneActions).to receive(:text!)
+      phone_number.text("hello world!")
```
Comment (johnzw89) [inline phone_number_spec.rb:22]: hitting the method is good to make sure there are no code errors. However, we should be explicit and test for a specific response. In this case it looks like the output should be json so we should test that we are getting the expected json response.

As mentioned above, we should be testing the API request component separately:
1.  Mock the API request. webmock/rspec is a good start to look into.
2. Stub the API request for failure and success
3. Test that your code correctly returns the success and fail responses.

File: phone_number.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  USERNAME = ENV['ACTIONS_USERNAME']
+  PASSWORD = ENV['ACTIONS_PASSWORD']
+
+  def initialize(number)
+    @number = number
+    @number = format_number
+  end
+
+  def text(message)
+    PhoneActions.new(USERNAME, PASSWORD).text!(@number, message)
```
Comment (johnzw89) [inline phone_number.rb:13]: not: consider saving PhoneActions.new(USERNAME, PASSWORD) as a instance variable on initialization. We don't need to initialize new objects on each method call.

**Review by johnzw89 — CHANGES_REQUESTED** [review]
Overall a great start to the phone number and actions feature. I left some feedback on how this can refactored to make more readable.

Biggest changes are around 
- handling API requests responses
- handling edge cases for a phone number
- adding more tests to test for the above. This might be complicated to setup so happy to pair.
